Library/Homebrew/software_spec.rb

```diff
@@ -53,7 +53,7 @@ def owner=(owner)
     @resource.owner = self
     resources.each_value do |r|
       r.owner     = self
-      r.version ||= version
+      r.version ||= version.dup
```

Comment:
That's why I think we should only call `dup` for non-Head version, which doesn't have commit.